File: Server/Models/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Server.Models
{
    public class Category
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [Required]
        [StringLength(50)]
        public string IconName { get; set; }

        public int? ParentCategoryId { get; set; }

        [ForeignKey("ParentCategoryId")]
        public Category ParentCategory { get; set; }

        public ICollection<Category> Categories { get; set; }
        public ICollection<Equipment> Equipment { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Client/src/types/Category.ts
export interface Category {
  id: number;
  name: string;
  slug: string;
  iconName?: string;
  parentCategoryId?: number;
  parentCategory?: Category;
  subcategories?: Category[];
  createdAt?: string;
  updatedAt?: string;
}

export interface CategoryEdit {
  name: string;
  slug: string;
  iconName: string;
  description?: string;
  parentCategoryId?: number;
  parentCategory?: Category;
  subcategories?: Category[];
}
